### PR TITLE
samples: nrf9160: agps: remove sample

### PR DESCRIPTION
**Pull request by tokangas**
Removed the A-GPS sample. nRF Cloud A-GPS and P-GPS are demonstrated in the GNSS sample (PR #6010), so there's no need for a separate A-GPS sample.

Added DNM label until PR #6010 has been merged.

### CODE REVIEW

File: doc/nrf/ug_nrf91_features.rst [hunk 1]
```diff
@@ -348,7 +348,6 @@ Following is a list of the samples and applications with some information about
 
 * :ref:`asset_tracker_v2` application - Uses nRF Cloud for A-GPS or P-GPS  or a combination of both. The application obtains GNSS fixes and transmits them to nRF Cloud along with sensor data.
 * :ref:`serial_lte_modem` application - Uses AT commands to start and stop GNSS and has support for nRF Cloud A-GPS and P-GPS. The application displays tracking and GNSS fix information in the serial console.
-* :ref:`agps_sample` sample - Uses nRF Cloud for A-GPS by default and can be configured to use SUPL. The sample obtains GNSS fixes and transmits them to nRF Cloud.
 * :ref:`gps_with_supl_support_sample` sample - Does not use A-GPS by default but can be configured to use SUPL. The sample displays tracking, fix information, and NMEA strings in the serial console.
```
Comment (peknis) [inline doc/nrf/ug_nrf91_features.rst:351]: Are we not adding the gnss_sample here?

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_nrf91_features.rst (hunk 1 above)
Comment (tokangas) [reply doc/nrf/ug_nrf91_features.rst:351]: That's in PR #6010.

Comment (peknis) [reply doc/nrf/ug_nrf91_features.rst:351]: Of course. My bad.

**Comment by plskeggs** [comment]
"nRF Cloud A-GPS and P-GPS are demonstrated in the GNSS sample (PR #6010), so there's no need for a separate A-GPS sample."  This is not quite correct, in that the older sample uses MQTT but the newer uses REST.  The distinction is important.

Isn't it unusual to simply remove a sample without any deprecation warning?  I thought the policy was to deprecate for 1 release and remove at the following release.

**Comment by tokangas** [comment]
> Isn't it unusual to simply remove a sample without any deprecation warning? I thought the policy was to deprecate for 1 release and remove at the following release.

My understanding is that there's no such policy for samples. @rlubos, is it ok to remove a sample without deprecating it first?

**Comment by tokangas** [comment]
@peknis There are references to `agps_sample` in old release notes. As we already discussed, when the sample is removed, those links still need to point somewhere.

**Comment by peknis** [comment]
> @peknis There are references to `agps_sample` in old release notes. As we already discussed, when the sample is removed, those links still need to point somewhere.

Yep, I'm discussing this with my colleague to come up with a proposal. Hold on.

**Comment by rlubos** [comment]
> > Isn't it unusual to simply remove a sample without any deprecation warning? I thought the policy was to deprecate for 1 release and remove at the following release.
> 
> My understanding is that there's no such policy for samples. @rlubos, is it ok to remove a sample without deprecating it first?

I think so, with library deprecation makes sense as it might be used by some customer project, so the deprecation gives the3rd party application time to adapt. But for sample, I don't see a need for a deprecation period. Especially that the functionality is exposed in a `gnss` sample now, I don't see a reason to carry the old sample around.

**Review by rlubos — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**